### ISSUE TITLE
What's the reason for using a "develop" branch?

### ISSUE THREAD

**Issue by choldgraf** [ExecutableBookProject/MyST-Parser#21]
Curious why we are calling it "develop" and not just using "master" since this is our own repository and not a fork.

I think the name itself is kind of trivial, but remembering to `git checkout develop` instead of `master` feels like an unnecessary cognitive overhead to remember. I'm just curious if there's a specific reason for this, or plans to change it to `master` at some point?

**Comment by chrisjsewell** [comment]
I would point you towards: https://github.com/aiidateam/aiida-core/wiki/Branch-descriptions and this diagram of the git model 😄 

![git model](https://nvie.com/img/git-model@2x.png)

**Comment by choldgraf** [comment]
Haha - let me rephrase my point:

The use of development branches in general is definitely useful in organizing complex projects where you value stability in the master branch.

However, I don't think they are the standard pattern of operation in the scientific python community, and so they'll create cognitive burden for anybody from that community that works on this project (including me :-P). E.g. [numpy](https://github.com/numpy/numpy), [matplotlib](https://github.com/matplotlib/matplotlib), [pandas](https://github.com/pandas-dev/pandas) (I could go on) all use the "master branch contains the latest unstable version of code, and tags or release branches contain releases" pattern.

While I agree that in a sense using a "develop" branch is a best-practice, in the name of adhering to community standards I think it makes more sense to use "master". WDYT?

**Comment by chrisjsewell** [comment]
I'm +1 for a develop branch, but I guess it depends if the majority want to overall lol @jstac  @mmcky

**Comment by jstac** [comment]
I'm with @choldgraf on this one --- right or wrong, my understanding is that the convention he suggests will make it easier for others from our target audience to contribute, since they're more used to that pattern.

**Comment by chrisjsewell** [comment]
Ok well I'm not going to go down that hole of transitioning today. I imagine/hope that you can just delete the master branch and rename the develop branch to master

**Comment by choldgraf** [comment]
@chrisjsewell yeah for sure - it's not tough to switch them over...it's the kind of thing that is a pain if the project has been around for 10 years because everyone has to update their forks, but it's not a big deal to do for one repo

**Comment by choldgraf** [comment]
Are folks OK with me doing this today or tomorrow? I keep accidentally doing `git checkout master` and breaking my myst_parser :-P

**Comment by mmcky** [comment]
I'm in favour of it. @chrisjsewell is probably still travelling back to the UK. So we should send him a notification once its done.

**Comment by choldgraf** [comment]
I wanna make sure @chrisjsewell gives his blessing first :-)

**Comment by chrisjsewell** [comment]
Ok, this is done, but as a compromise I'm invoking #53 😉 as the package is pretty stable now

**Comment by choldgraf** [comment]
@chrisjsewell nice compromise, I like the idea :-)
